x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/categorization/TokenListCategorizer.java

```diff
@@ -40,6 +42,25 @@
  */
 public class TokenListCategorizer implements Accountable {
 
+    /**
+     * TokenListCategorizer that takes ownership of the CategorizationBytesRefHash and releases it when closed.
+     */
+    public static class CloseableTokenListCategorizer extends TokenListCategorizer implements Releasable {
```

Comment:
👍

Reply:
FYI: this was needed, because the autogen code automatically closes releasables. However, the original `TokenListCategorizer` doesn't take ownership of the ByteRef, leading to leaked memory (and failing tests; kudos for creating those tests)